versioned_docs/version-8.6/self-managed/setup/guides/secret-management.md

```diff
@@ -0,0 +1,356 @@
+| ------------------------------------ | --------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------- | ----------------------- |
+| **External Database Password**       | `webModeler.restapi.externalDatabase.existingSecret`, `webModeler.restapi.externalDatabase.existingSecret.name` | Password for external PostgreSQL if using an external DB | Not set unless provided |
+| **SMTP Password**                    | `webModeler.restapi.mail.existingSecret`                                                                        | SMTP credentials for sending email notifications         | Not set unless provided |
+| **Connectors Inbound Auth Password** | `connectors.inbound.auth.existingSecret`, `connectors.inbound.auth.existingSecretKey`                           | Basic auth password for Connectors polling Operate       | Not set unless provided |
+
+### External Search (Elasticsearch / OpenSearch)
+
+| **Secret**                          | **Chart Values Key**                       | **Purpose**                                                     | **Default Behavior**    |
+| ----------------------------------- | ------------------------------------------ | --------------------------------------------------------------- | ----------------------- |
+| **External Elasticsearch Auth**     | `global.elasticsearch.auth.existingSecret` | Password for external Elasticsearch authentication (basic auth) | Not set unless provided |
+| **External Elasticsearch TLS Cert** | `global.elasticsearch.tls.existingSecret`  | TLS certificate for external Elasticsearch over SSL             | Not set unless provided |
+| **External OpenSearch Auth**        | `global.opensearch.auth.existingSecret`    | Password for external OpenSearch authentication (basic auth)    | Not set unless provided |
+| **External OpenSearch TLS Cert**    | `global.opensearch.tls.existingSecret`     | TLS certificate for external OpenSearch over SSL                | Not set unless provided |
+
+---
+
+## Referencing Secrets in `values.yaml`
+
+> ✅ **Best Practice:** Use the structured form (`existingSecret.name` and `existingSecretKey`) when working with internal secrets. It allows more precise control over which keys are used and promotes reuse of secrets across components.
+
+There are two primary patterns for referencing secrets, depending on the chart and component:
+
+- Use `existingSecret` (flat string) for **simple secrets** such as external database passwords or SMTP credentials.
+- Use `existingSecret.name` and `existingSecretKey` (structured form) for **internal secrets** like OAuth client passwords, where the chart expects to access a specific key inside the secret.
+
+> ℹ️ If a field supports the structured format, it's generally safer and clearer to use it.
+> The structured format also enables more consistent secret reuse across multiple components.
+
```

Comment:
Use Use the correct format to avoid configuration errors:
- Flat string (`existingSecret`): For simple references like external DBs or SMTP.
- Structured (`existingSecret.name` and `existingSecretKey`): Recommended for internal secrets and OAuth client to access a specific key reference.